app/presenters/bill-runs/setup/year.presenter.js

```diff
@@ -2,23 +2,75 @@
 
 /**
  * Formats data for the `/bill-runs/setup/{sessionId}/year` page
- * @module RegionPresenter
+ * @module YearPresenter
  */
 
+const LicenceSupplementaryYearModel = require('../../../models/licence-supplementary-year.model.js')
+
 /**
  * Formats data for the `/bill-runs/setup/{sessionId}/year` page
  *
  * @param {module:SessionModel} session - The session instance to format
  *
  * @returns {object} - The data formatted for the view template
  */
-function go (session) {
+async function go (session) {
+  const selectedYear = session.year ? session.year : null
+
+  let financialYearsData = []
+
+  if (session.type === 'two_part_tariff') {
+    financialYearsData = _annualFinancialYearsData(selectedYear)
+  }
+
+  if (session.type === 'two_part_supplementary') {
+    financialYearsData = await _supplementaryFinancialYearsData(session.region, selectedYear)
+  }
+
   return {
+    financialYearsData,
     sessionId: session.id,
-    selectedYear: session.year ? session.year : null
+    selectedYear
   }
 }
 
+function _annualFinancialYearsData (selectedYear) {
+  return [
+    { text: '2023 to 2024', value: '2024', checked: selectedYear === '2024' },
+    { text: '2022 to 2023', value: '2023', checked: selectedYear === '2023' },
+    { text: '2021 to 2022', value: '2022', checked: selectedYear === '2022' },
+    { text: '2020 to 2021', value: '2021', checked: selectedYear === '2021' }
+  ]
+}
+
+async function _supplementaryFinancialYearsData (regionId, selectedYear) {
+  const supplementaryFinancialYearsData = []
+  const tptSupplementaryYears = await _tptSupplementaryYears(regionId)
+
+  if (tptSupplementaryYears.length > 0) {
+    tptSupplementaryYears.forEach((tptSupplementaryYear) => {
+      const { financialYearEnd } = tptSupplementaryYear
+
+      supplementaryFinancialYearsData.push({
+        text: `${financialYearEnd - 1} to ${financialYearEnd}`,
+        value: financialYearEnd.toString(),
+        checked: selectedYear === financialYearEnd.toString()
+      })
+    })
+  }
+
+  return supplementaryFinancialYearsData
+}
+
+async function _tptSupplementaryYears (regionId) {
+  return LicenceSupplementaryYearModel.query()
```

Comment:
This is more of a question than a change but we don't fetch data in any of our other presenters, we do that in a separate service normally and then pass the fetched data to the presenter, is this the exception to that rule?

Reply:
I have to use it in both the GET & POST requests which is why I popped it in the presenter 🤠 I'm going to move it out of the presenter into its own service.